libs/blocks/aside/aside.js

```diff
@@ -46,32 +51,130 @@ function decorateStaticLinks(el) {
 }
 
+function decorateModalImage(el, lnkImg) {
+  if (el.classList.contains('split')) lnkImg.closest('div').classList.add('has-modal');
+  lnkImg.classList.add('play-btn');
+  const playIcon = createTag('div', { class: 'play-icon-container', 'aria-label': 'play' }, PLAY_ICON);
+  const playCircle = createTag('div', { class: 'play-btn-circle', 'aria-label': 'play' }, playIcon);
+  const playContainer = createTag('div', { class: 'play-container', 'aria-label': 'play' });
+  lnkImg.parentNode.appendChild(lnkImg.querySelector('picture'));
+  lnkImg.parentNode.appendChild(playContainer);
+  lnkImg.appendChild(playCircle);
+  playContainer.appendChild(lnkImg);
+}
+
+function decorateIconStack(el) {
+  if (!(el.classList.contains('medium') || el.classList.contains('large'))) return;
+  const stackEl = el.querySelector('ul img');
+  if (stackEl) {
+    stackEl.closest('ul').classList.add('icon-stack-area', 'body-s');
+    el.classList.add('icon-stack');
+  }
+  const stackItems = el.querySelectorAll('ul li');
+  [...stackItems].forEach((stackItem) => {
+    const links = stackItem.querySelectorAll('a');
+    if (stackItem.querySelectorAll('a').length <= 1) return;
+    const picIndex = links[0].querySelector('a picture') ? 0 : 1;
+    const linkImg = links[picIndex];
+    const linkText = links[1 - picIndex];
+    linkText.prepend(linkImg.querySelector('picture'));
+    linkImg.remove();
+  });
+}
+
+function decorateMedia(el) {
+  if (!(el.classList.contains('medium') || el.classList.contains('large'))) return;
+  const allMedia = el.querySelectorAll('div > p video, div > p picture');
+  let processed = false;
+  allMedia.forEach((media) => {
+    if (processed) return;
+    const parentP = media.closest('p');
+    const siblingP = parentP?.nextElementSibling;
+    if (!siblingP || siblingP.nodeName.toLowerCase() !== 'p') return;
+    const siblingText = siblingP.textContent;
+    const hasFormats = siblingText.match(FORMAT_REGEX)?.index;
+    if (!(hasFormats === 0)) return;
+    processed = true;
+    const formats = siblingText.split(': ')[1]?.split(/\s+/);
+    const formatClasses = formats ? ['format',
+      `desktop-${formats[((formats.length - 1) > 0) ? (formats.length - 1) : 0]}`,
+      `tablet-${formats[((formats.length - 2) > 0) ? (formats.length - 2) : 0]}`,
+      `mobile-${formats[0]}`,
+    ] : [];
+    media.closest('div').classList.add(...formatClasses);
+    siblingP.remove();
+    media.closest('div').insertBefore(media, parentP);
+    parentP.remove();
+  });
+}
+
+function decorateVideo(container) {
+  const link = container.querySelector('a[href*=".mp4"]');
+  if (!link) return;
+  const isNotLooped = !!(link.hash?.includes('autoplay1'));
+  const attrs = `playsinline autoplay ${isNotLooped ? '' : 'loop'} muted`;
+  container.innerHTML = `<video preload="metadata" ${attrs}>
+    <source src="${link.href}" type="video/mp4" />
+  </video>`;
+  container.classList.add('has-video');
+}
+
+function decorateBlockBg(block, node) {
```

Comment:
Is there a reason this util/decorate func() is brought in here? 
To keep things D.R.Y. and preventing duplicate code, can we update the `/milo/libs/utils/decorate.js` instance of this function and keep it as an import instead?. The above `decorateVideo()` would also likely also move. 
Also I imagine other blocks would benefit from this change, but would need tested. (aside, marquee, marquee-anchors, media, text, z-pattern) are using decorateBlockBg()

Reply:
@ryanmparrish decorateBlockBg was brought in as asides as the video background support is requested for the asides. 
This was made to be similar to Marquee which has its own decorateBlockBg which provides multiple support like video and focal point but were not required for other blocks.

Reply:
> This was made to be similar to Marquee which has its own decorateBlockBg

Im good w/ this for now, but as more blocks need the same features I'd love to have a discussion around moving these back into a shared util import and extending that func() rather than near duplication across multiple places. Ideally we nip the bud now and not add tech debt IMO.